src/otp-verification/types.ts

```diff
@@ -0,0 +1,152 @@
+import { BaseFormElementProps } from "src/form/types";
+import { PhoneNumberInputValue } from "src/phone-number-input";
+import { useCountdown } from "../util";
+
+export enum InternalOtpState {
+    DEFAULT = "default",
+    SENT = "sent",
+    VERIFIED = "verified",
+}
+
+export enum OtpVerifyType {
+    EMAIL = "email",
+    PHONE_NUMBER = "phone-number",
+}
+
+// Base props for OTP verification component
+export interface BaseOtpVerificationProps {
+    id?: string | undefined;
+    "data-testid"?: string | undefined;
+    disabled?: boolean;
```

Comment:
add union to optional props per repo convention

```suggestion
    disabled?: boolean | undefined;
```